dali/pipeline/data/type_traits.h

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) 2020, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_PIPELINE_DATA_TYPE_TRAITS_H_
+#define DALI_PIPELINE_DATA_TYPE_TRAITS_H_
+
+namespace dali {
+
+template <typename Backend>
+class TensorVector;
+
+template <typename Backend>
+class TensorList;
+
+class CPUBackend;
+class MixedBackend;
+class GPUBackend;
+
+template <typename Backend>
+struct is_backend {
```

Comment:
Maybe?
```c++
template <typename Backend>
struct is_backend {
  static constexpr bool value = 
    std::is_same<Backend, CPUBackend>::value || std::is_same<Backend, MixedBackend>::value ||
    std::is_same<Backend, GPUBackend>::value;
};
```
It seems like a tool that could be reused so we should make it more like standard traits.

Also this way of writing it imposes some inconvenience: 
```c++
bool b = is_backend<int>();   // <-- won't compile, cannot convert
```

Reply:
Done